src/renderer/contexts/GlobalNodeState.tsx

```diff
@@ -445,23 +445,46 @@ export const GlobalProvider = memo(
         }, [hasRelevantUnsavedChanges, changeNodes, changeEdges, setSavePath, setViewport]);
 
         const performSave = useCallback(
-            (saveAs: boolean) => {
+            (saveAs: boolean, clearInputData = false, saveSeparately = false) => {
                 (async () => {
                     try {
                         const saveData = dumpState();
+                        if (clearInputData) {
+                            saveData.nodes = saveData.nodes.map((n) => {
+                                const inputData = { ...n.data.inputData } as Mutable<InputData>;
+                                const nodeSchema = schemata.get(n.data.schemaId);
+                                nodeSchema.inputs.forEach((input) => {
+                                    if (['file', 'directory'].includes(input.kind)) {
```

Comment:
This array is only typed as `string`.
```suggestion
                                    const clearKinds = new Set<InputKind>(['file', 'directory']);
                                    if (clearKinds.has(input.kind)) {
```

Also, since the `clearKinds` set could be moved outside the `nodes.map`.